alica_engine/include/engine/scheduler/Scheduler.h

```diff
@@ -66,9 +69,12 @@ class Scheduler
 
     void terminate()
     {
-        _running = false;
-
+        {
+            std::unique_lock<std::mutex> lock(_mutex);
+            _running = false;
+        }
         _cv.notify_one();
```

Comment:
Hm, i think you might still loose the `_cv` notification on around line 100 before the lock. Maybe then you need to check the values before the `wait()`.

Reply:
that's okay, since `running` will be false in that case & we will never wait on the condition variable

Reply:
the predicate passed to the wait() function is called before going into wait but under the lock

Reply:
right